src/Processors/Formats/Impl/AvroRowInputFormat.cpp

```diff
@@ -631,6 +665,42 @@ AvroDeserializer::Action AvroDeserializer::createAction(const Block & header, co
         }
         return AvroDeserializer::Action::unionAction(branch_actions);
     }
+    else if (node->type() == avro::AVRO_ARRAY)
+    {
+        /// If header doesn't have column with current_path name and node is Array(Record),
+        /// check if we have a flattened Nested table with such name.
+        Names nested_names = Nested::getAllNestedColumnsForTable(header, current_path);
+        auto nested_avro_node = node->leafAt(0);
```

Comment:
Can't we have empty node here, so is it safe to call `leafAt(0)` without checking `node->leaves()`?

Reply:
We checked that node type is Array, so it should have one leaf

Reply:
Hm, but it's interesting if it's possible to corrupt data in this way

Reply:
Tried to create such corrupted file, but without success, reading such file failes in avro library before this code